packages/peregrine/lib/List/useListState.js

```diff
@@ -218,7 +225,13 @@ const getInitiallySelectedKeys = ({
             return initialSelection.map(getItemKey);
         }
 
-        return [getItemKey(initialSelection)];
+        const itemKey = getItemKey(initialSelection);
+
+        if (itemKey) {
+            return [itemKey];
+        }
+
+        return [];
```

Comment:
This didn't actually manifest in a bug that I could tell but it was initializing with an array of `[undefined]` which seems wrong. If there is no `itemKey` that should be handled here as a case rather than the former method.